### PR TITLE
feat(GitOpsCluster): Propagate ManagedCluster labels to created secrets

### LINKED ISSUE #50: Labels set on ManagedClusters should be propagated to ArgoCD
The labels set on a ManagedCluster should be propagated to ArgoCD, so they can be used in ArgoCD to e.g. select different Kustomize overlays in the same `ApplicationSet`.

### PR DESCRIPTION
**Pull request by 0xFelix**
This propagates labels set on ManagedClusters to the secrets created by the GitOpsCluster controller. The labels can e.g. be used in ArgoCD to select different environments when using Kustomize.

The reconciliation code of the GitOpsCluster controller is changed so secrets are always updated, whether they are blank secrets or not. This allows to propagate changes to the labels of a ManagedCluster after a secret was already created.

Fixes #50

### CODE REVIEW

File: pkg/controller/gitopscluster/gitopscluster_controller.go [hunk 1]
```diff
@@ -656,14 +671,31 @@ func (r *ReconcileGitOpsCluster) GetManagedClusters(namespace string, placementr
...
+			for i, cluster := range clusterList.Items {
+				if cluster.Name == clusterDecision.ClusterName {
+					klog.Info("found managed cluster: " + clusterDecision.ClusterName)
+
+					clusters = append(clusters, clusterList.Items[i])
+
+					break
+				}
+			}
 		}
```
Comment (xiangjingli) [inline pkg/controller/gitopscluster/gitopscluster_controller.go:699]: There could be a performance issue. e.g.  if the whole number of managed cluster is 3K (that is the scenario that ACM supports now),  for each matched cluster name in the placementDecision list, it needs 3k loop to find its managedCluster.

I would suggest to save all managed Cluster manifest to a map for later faster search/match, where the cluster name is the key of the map so sth like this:

err = r.List(context.TODO(), clusterList, &client.ListOptions{})
if err != nil {
    klog.Error("Failed to list managed clusters, err:", err)
    return nil, err
}

clusterMap := make(map[string]spokeclusterv1.ManagedCluster)
for _, managedCluster := range clusterList.Items {
    clustermap[managedCluster.Name] =  managedCluster
}

Then in the later placementDecision loop

for _, clusterDecision := range placementdecision.Status.Decisions {
    if cluster, ok :=  clustermap[clusterDecision.ClusterName] {
        clusters = append(clusters, cluster)
    }
}

**Comment by xiangjingli** [comment]
@0xFelix  Thanks Felix for the contribution!   The rest looks good!

@lennysgarage  Please double review the PR when you have a cycle. Make sure it won't break the union secret feature you implemented before.

**Comment by lennysgarage** [comment]
@xiangjingli Feature looks good! It shouldn't cause any problems with the union secret data as we union the labels after they have all been collected.

File: pkg/controller/gitopscluster/gitopscluster_controller.go (hunk 1 above)
Comment (0xFelix) [reply pkg/controller/gitopscluster/gitopscluster_controller.go:700]: Very good suggestion, I wasn't content with this implementation too. I updated it to follow your suggestion.

**Pull request merged**